src/w_wad.c

```diff
@@ -495,40 +495,45 @@ void WritePredefinedLumpWad(const char *filename)
    if((file = M_fopen(fn, "wb")))
    {
       wadinfo_t header = { "PWAD" };
-      size_t filepos = sizeof(wadinfo_t) + num_predefined_lumps * sizeof(filelump_t);
+      size_t filepos = sizeof(wadinfo_t) + num_lumps * sizeof(filelump_t);
       int i;
       
-      header.numlumps     = LONG(num_predefined_lumps);
+      header.numlumps     = LONG(num_lumps);
       header.infotableofs = LONG(sizeof(header));
       
       // write header
       fwrite(&header, 1, sizeof(header), file);
       
       // write directory
-      for(i = 0; i < num_predefined_lumps; i++)
+      for(i = 0; i < num_lumps; i++)
       {
          filelump_t fileinfo = { 0 };
          
          fileinfo.filepos = LONG(filepos);
-         fileinfo.size    = LONG(predefined_lumps[i].size);         
-         M_CopyLumpName(fileinfo.name, predefined_lumps[i].name);
+         fileinfo.size    = LONG(lumps[i].size);
+         M_CopyLumpName(fileinfo.name, lumps[i].name);
          
          fwrite(&fileinfo, 1, sizeof(fileinfo), file);
 
-         filepos += predefined_lumps[i].size;
+         filepos += lumps[i].size;
       }
       
       // write lumps
-      for(i = 0; i < num_predefined_lumps; i++)
-         fwrite(predefined_lumps[i].data, 1, predefined_lumps[i].size, file);
+      for(i = 0; i < num_lumps; i++)
+         fwrite(lumps[i].data, 1, lumps[i].size, file);
       
       fclose(file);
-      I_Error("Predefined lumps wad, %s written, exiting\n", filename);
+      I_Error("\nInternal lumps wad, %s written, exiting\n", filename);
```

Comment:
Why error?

Reply:
I guess the engine is supposed to exit with a message when being "misused" an a data exporter.

Reply:
Won't this cause a nonzero exit status? Just thinking this abuse probably causes problems for people writing scripts, but it would need to be addressed separately anyway.

Reply:
Hm, dammit!